dev/io.openliberty.cdi.4.0.internal_fat/fat/src/io/openliberty/cdi40/internal/fat/config/LegacyConfigTest.java

```diff
@@ -33,60 +39,70 @@
                                                       .addClass(RequestScopedBean.class);
         CDIArchiveHelper.addEmptyBeansXML(legacyEmptyBeansXMLWar);
         ShrinkHelper.exportDropinAppToServer(server, legacyEmptyBeansXMLWar, DeployOptions.SERVER_ONLY);
 
-        //This legacyBeans10XMLWar war contains a non-empty CDI 1.0 beans.xml. This means that when using CDI 4.0, it will default to a discovery mode of
-        //ANNOTATED. The legacy configuration option has no effect. At first we thought that it should have been treated the same an empty
-        //beans.xml such that the config option would have maintained legacy behaviour ... but this is not so and is consistent with the
-        //spec (which only mentions empty beans.xml files).
-        WebArchive legacyBeans10XMLWar = ShrinkWrap.create(WebArchive.class, LEGACY_BEANS10_APP_NAME + ".war")
-                                                   .addClass(AnnotatedBeansServlet.class)
-                                                   .addClass(UnannotatedBean.class)
-                                                   .addClass(RequestScopedBean.class);
-        CDIArchiveHelper.addBeansXML(legacyBeans10XMLWar, CDIVersion.CDI10);
-        ShrinkHelper.exportDropinAppToServer(server, legacyBeans10XMLWar, DeployOptions.SERVER_ONLY);
+        //This application contains a non-empty CDI 1.0 beans.xml. When using CDI 4.0, the discovery mode will default to
+        //ANNOTATED. Unlike an empty beans.xml, the legacy configuration option has no effect. Therefore we will output a warning when the
+        //DeploymentException occurs
+        WebArchive unannotatedlegacyBeans10XMLWar = ShrinkWrap.create(WebArchive.class, LEGACY_BEANS10_APP_NAME + ".war")
+                                                              .addClass(AllBeansServlet.class)
+                                                              .addClass(UnannotatedBean.class)
+                                                              .addClass(RequestScopedBean.class);
+        CDIArchiveHelper.addBeansXML(unannotatedlegacyBeans10XMLWar, CDIVersion.CDI10);
+        ShrinkHelper.exportDropinAppToServer(server, unannotatedlegacyBeans10XMLWar, DeployOptions.SERVER_ONLY, DeployOptions.DISABLE_VALIDATION);
 
-        ServerConfiguration config = server.getServerConfiguration();
-        ConfigElementList<Cdi> cdis = config.getCdi();
-        Cdi cdi = new Cdi();
-        cdi.setEmptyBeansXMLExplicitBeanArchive(true);
-        cdis.add(cdi);
+        server.startServer();
+    }
 
-        server.updateServerConfiguration(config);
+    @Test
+    public void testUnversionedBeansXmlWarning() throws Exception {
+        //check for a warning message about a non-empty unversioned beans.xml file
+        List<String> warningMessages = server.findStringsInLogs("CWOWB1018W: .*[wsjar:file:.*" + LEGACY_BEANS10_APP_NAME + ".war!/WEB-INF/beans.xml]");
```

Comment:
`findStringsInLogs` takes a regular expression so you need to escape the `[` and `]` characters (or just remove them)

Reply:
done